### PR TITLE
Pw/rpc-sdk-server

### PR DESCRIPTION
**Pull request by piotrwitek**
Implemented client server communication with mock integration
Added serialization logic to domain objects implementations
Implemented refinance client server communication flow e2e test

Important to review:
Just review sdk-common, the rest was just moved around as was without much changes.

### CODE REVIEW

File: apps/summerfi-api/lib/sdk-router-function/tsconfig.test.json [hunk 1]
```diff
@@ -2,12 +2,14 @@
   "extends": "@summerfi/typescript-config/tsconfig.test.json",
   "compilerOptions": {
     "rootDir": ".",
-    "outDir": "dist",
     "baseUrl": ".",
     "paths": {
-      "~src/*": ["./src/*"]
+      "~src/*": ["src/*"],
+      "@summerfi/common": ["node_modules/@summerfi/common/src/index.ts"],
```
Comment (robercano) [inline apps/summerfi-api/lib/sdk-router-function/tsconfig.test.json:8]: I think this is what you were explaining about the solution to the imports for tests. If this is going to be the way of doing it, I think we should make a proposal so every other packages uses the same approach. I fear that we have a mix of configurations at this moment and that it is still not clear which one is best

File: sdk/sdk-common/src/common/implementation/Token.ts [hunk 2]
```diff
@@ -42,7 +51,27 @@ export class Token implements Printable {
     return new Token(params)
   }
 
-  public toString(): string {
+  deserialize(params: TokenSerialized): Token {
```
Comment (robercano) [inline sdk/sdk-common/src/common/implementation/Token.ts:54]: `static deserialize` ?

File: sdk/sdk-common/src/users/implementation/client/PositionClientImpl.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { RiskRatio, TokenAmount } from '~sdk-common/common'
```
Comment (robercano) [inline sdk/sdk-common/src/users/implementation/client/PositionClientImpl.ts:1]: As discussed in the meeting this morning, I think we will have the following pattern for classes:
`IPosition` -> Interface with all data members
`Position` -> Class extending `IPosition` and adding functionality

I think the `PositionClientImpl` and the server side of it won't exist anymore as this class does not need to communicate with a server counter part to retrieve anything, while classes like `User` do need to communicate with the server and will have the Client/Server parts.

Comment (robercano) [reply sdk/sdk-common/src/users/implementation/client/PositionClientImpl.ts:1]: Again, I know we discussed it this morning and I'm fine with not changing it in this PR and do it in the serialization PR, just pointing it out so we don't forget :)

File: sdk/swap-service/tests/oneinch.spec.ts [hunk 4]
```diff
@@ -3,8 +3,8 @@ import { Address, Percentage, Token, TokenAmount } from '@summerfi/sdk-common/co
 import { subtractPercentage } from '@summerfi/sdk-common/utils'
 import { getSwapManager, SwapData, SwapProviderType } from '~swap-service'
 
-describe.skip('OneInch | SwapManager | Integration', () => {
-  it('should provide swap data', async () => {
+describe('OneInch | SwapManager | Integration', () => {
+  it.skip('should provide swap data', async () => {
```
Comment (robercano) [inline sdk/swap-service/tests/oneinch.spec.ts:7]: I think Jacob also disabled this on `dev`. Just in case it conflicts when you merge

**Review by robercano — APPROVED** [review]

File: apps/summerfi-api/lib/sdk-router-function/tsconfig.test.json (hunk 1 above)
Comment (piotrwitek) [reply apps/summerfi-api/lib/sdk-router-function/tsconfig.test.json:8]: Yes, I think this is a way to go as it turns out we cannot share path mappings and each tsconfig test needs to have own mapping as it overrides when extending base tsconfig, no way around it!
I'll refactor all existing packages to have a common approach.

File: sdk/sdk-common/src/common/implementation/Percentage.ts [hunk 5]
```diff
@@ -15,11 +19,19 @@ export class Percentage implements Printable {
...
+
   toString(): string {
     return `${this.value}`
   }
 
   add(percentage: Percentage): Percentage {
     return Percentage.createFrom({ percentage: this.value + percentage.value })
   }
+
+  serialize(): PercentageSerialized {
```
[the remark this answers is not in the document]
Comment (piotrwitek) [reply sdk/sdk-common/src/common/implementation/Percentage.ts:34]: Agreed, I'll remove those in the new approach PR as we agreed.

File: sdk/sdk-common/src/common/implementation/Token.ts [hunk 6]
```diff
@@ -32,7 +41,7 @@ export class Token implements Printable {
     this.decimals = params.decimals
   }
 
-  public static createFrom(params: {
```
[the remark this answers is not in the document]
Comment (piotrwitek) [reply sdk/sdk-common/src/common/implementation/Token.ts:35]: Yes public is default no need to add it ;) Thanks for pointing out I'll clean that out.

File: sdk/sdk-common/src/users/implementation/client/PositionClientImpl.ts (hunk 3 above)
Comment (piotrwitek) [reply sdk/sdk-common/src/users/implementation/client/PositionClientImpl.ts:1]: Thanks for adding your expectation clearly here, I'll see if I can do that and it make sense if not I'll skip as suggested :)

**Review by paszkowskiDamian — APPROVED** [review]

**Pull request merged**